### PR TITLE
Added Content Hub and URLdispatcher guides

### PR DESCRIPTION
**Pull request by cibersheep**

### CODE REVIEW

File: appdev/guides/contenthub.rst [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+.. figure:: /_static/images/appdev/guides/contenthubimages/01.png
+        :align: center
+
+        Content Hub Share Page
+
+On Ubuntu Touch the apps are confined. The way of sharing files between them is through the Content Hub, a part of the system that takes care of file import, export and sharing.
+
+Different ways of sharing the content
+-------------------------------------
+As we can see in the `documentation <https://api-docs.ubports.com/sdk/apps/qml/Ubuntu.Content/ContentHandler.html#detailed-description>`_, there are several ways of handling the file to be shared:
```
Comment (UniversalSuperBox) [inline appdev/guides/contenthub.rst]: I prefer giving links a descriptive name, so here I would write `Content Hub documentation` as the link text.

File: appdev/guides/contenthub.rst [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+- ContentHandler.Source (The app selected is going to be the source of the file imported)
+- ContentHandler.Destination (The app selected is going to be the destination of the file exported)
+- ContentHandler.Share (The app selected is going to be the app from witch the file is going to be shared from)
+
+Importing
+^^^^^^^^^
+.. figure:: /_static/images/appdev/guides/contenthubimages/02.png
+        :align: center
+
+        Webapp Creator on the OpenStore.io
```
Comment (UniversalSuperBox) [inline appdev/guides/contenthub.rst]: I think we normally call the store `OpenStore` rather than `OpenStore.io`.

File: appdev/guides/contenthub.rst [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+        ContentPeerPicker {
+            anchors { fill: parent; topMargin: picker.header.height }
+            visible: parent.visible
+            showTitle: false
+            contentType: picker.contentType //ContentType.Pictures
+            handler: picker.handler //ContentHandler.Source
+
+**ContentPeerPicker** is the element that shows the apps.
+**contentType** is passed in the code as ContentType.Pictures. So, we will only see apps from which we only can import images.
```
Comment (UniversalSuperBox) [inline appdev/guides/contenthub.rst]: Looks like the code in the example doesn't actually pass the `ContentType`. Is that elsewhere in the code? If so, maybe we'd like to add it here.

File: appdev/guides/contenthub.rst [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+        Gelek in the OpenStore.io
+
+In Gelek, we are going to end with some saved games that we want to save in our device or share with ourselves (in Telegram and then save them to our computer).
+
+.. image:: /_static/images/appdev/guides/contenthubimages/05.png
+
+Tapping on the download icon we will get a Content Hub to «save» the game file (which is actually an export).
+
+The game file is a file type *.glksave.* So we will tell to the Content Hub that is of type «All» (`see code <https://gitlab.com/cibersheep/gelek/blob/master/app/InstallPage.qml#L38>`_).
```
Comment (UniversalSuperBox) [inline appdev/guides/contenthub.rst]: In the case of file extensions, code, or other non-translatable strings, it's better to place them in double backquotes (\`). This will make them appear monospaced in the output (which makes them easier to read) and hopefully it shows translators that they shouldn't translate.

File: appdev/guides/contenthub.rst [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            anchors { fill: parent; topMargin: picker.header.height }
+            visible: parent.visible
+            showTitle: false
+            contentType: ContentType.All
+            handler: ContentHandler.Destination
+
+            onPeerSelected: {
+
+**contentType** is *ContentType.All.* So, we will only see apps that are able to receive files of type other.
+And **handler** is *ContentHandler.Destination.* The app selected will be the destination of our saved game.
```
Comment (UniversalSuperBox) [inline appdev/guides/contenthub.rst]: Starting a sentence with "And" or "So" is generally not very clear in English. "The **contentType** `ContentType.All` will show the user apps which accept the type `other`. The **handler** `ContentHandler.Destination.*` tells Content Hub that the app the user selects is the destination of the file."

Or something like that. I feel like I changed the meaning of the sentence.

File: appdev/guides/contenthub.rst [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+                "videos",
+                "contacts",
+                "music"
+            ]
+        }
+
+So, with this example, uMatriks will be able to be the destination, source and share app for all kinds of ContentType.
+What about the other hooks in the manifest.json?
+
+That is «Flour from another sack» and will be discussed in the next article.
```
Comment (UniversalSuperBox) [inline appdev/guides/contenthub.rst]: This idiom doesn't work in English at all. :P

File: appdev/guides/contenthub.rst [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+
+
```
Comment (UniversalSuperBox) [inline appdev/guides/contenthub.rst]: Please only place one newline at the end of each file.

File: appdev/guides/importing-CH-urldispatcher.rst [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+
+Handle data from the Content Hub
+--------------------------------
+
+.. figure:: /_static/images/appdev/guides/importingCHdispatcherimages/02ichu.png
+        :align: center
+
+        OpenStore app from open-store.io
+
+One of the easiest ways of testing an app, is to send a test click to yourself, on Telegram, and opening that click file with the OpenStore through the Content Hub:
```
Comment (UniversalSuperBox) [inline appdev/guides/importing-CH-urldispatcher.rst]: No need for the second comma in this sentence.

File: appdev/guides/importing-CH-urldispatcher.rst [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+                    contentHubInstallInProgress = true;
+                    PlatformIntegration.clickInstaller.installPackage(filePath)
+                })
+            }
+        }
+
+If we tap on the OpenStore app, it will be opened and it will ask if we want to install the click file. Let’s take a look into `the code <https://github.com/UbuntuOpenStore/openstore-app/blob/master/openstore/Main.qml#L85>`_ to see how it is done:
+
+.. image:: /_static/images/appdev/guides/importingCHdispatcherimages/04ichu.png
+        :align: center
```
Comment (UniversalSuperBox) [inline appdev/guides/importing-CH-urldispatcher.rst]: Any need for the image as a code sample? I'd prefer another block of code to a screenshot...

Good developers copy. Great developers paste.

File: appdev/guides/contenthub.rst [hunk 10]
```diff
@@ -0,0 +1,157 @@
+=============================================
+The content hub. Tips and tricks
+=============================================
+.. figure:: /_static/images/appdev/guides/contenthubimages/01.png
```
[the remark this answers is not in the document]
Comment (cibersheep) [reply appdev/guides/contenthub.rst]: - Done
- Regular hyphen? Not en dash, right?

File: appdev/guides/contenthub.rst (hunk 2 above)
Comment (cibersheep) [reply appdev/guides/contenthub.rst]: Oh, that was for the Medium page... silly me

File: appdev/guides/contenthub.rst (hunk 3 above)
Comment (cibersheep) [reply appdev/guides/contenthub.rst]: Done

**Comment by cibersheep** [comment]
Let me know if there is something else you want me to review :)

File: appdev/guides/importing-CH-urldispatcher.rst (hunk 8 above)
Comment (UniversalSuperBox) [reply appdev/guides/importing-CH-urldispatcher.rst]: "One of the easiest ways of testing an app is to send a test click to yourself on Telegram, then open that click file with the OpenStore through the Content Hub"

File: appdev/guides/importing-CH-urldispatcher.rst [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+                }
+            }
+
+            //Start timer for Registering Status
+            checkStatus.start()
+        }
+
+
+All well and good but what happens if more than one app has the same URL type defined?
+--------------------------------------------------------------------------------------
```
Comment (UniversalSuperBox) [inline appdev/guides/importing-CH-urldispatcher.rst]: I'd shorten this to "What happens if more than one app has the same URL type defined?"

File: appdev/guides/importing-CH-urldispatcher.rst [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+
+A very good question. What happens if we tap on a Twitter link? How is such a URL handled by the `URLdispatcher` as protocol `http` or the protocol `http://twitter`?
+
+What happens if two apps have the same defined protocol?
+
+Now it’s time to do some tests and share the results in the next guide. At the same bat-hour, in the same bat-channel.
```
Comment (UniversalSuperBox) [inline appdev/guides/importing-CH-urldispatcher.rst]: Seems like a rather blog thing to say, doesn't make much sense in the docs. :D

File: appdev/guides/contenthub.rst [hunk 13]
```diff
@@ -0,0 +1,163 @@
...
+
+Wait a minute. Why the different apps?
+--------------------------------------
+
+.. figure:: /_static/images/appdev/guides/contenthubimages/08.png
+        :align: center
+
+        Content Hub: Export vs Share
+
+Each developer can decide the rules each app would follow, in relation to the Content Hub. Why the OpenStore is shown as the destination of an export?
```
Comment (UniversalSuperBox) [inline appdev/guides/contenthub.rst]: No need for the comma in this sentence

**Comment by cibersheep** [comment]
Is something missing :)?
